### PR TITLE
C++: 28 strsafe library updates2

### PR DESCRIPTION
**Pull request by bdrodes**
Added support for StrSafe.h sprintf variants.

### CODE REVIEW

File: cpp/ql/lib/change-notes/2023-11-8-strsafe-models.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+category: minorAnalysis
+---
+* Added missing models for strsafe sprintf variants.
```
Comment (MathiasVP) [inline cpp/ql/lib/change-notes/2023-11-8-strsafe-models.md:4]: How about something like this instead?

```suggestion
* Added models for the `sprintf` variants from the `StrSafe.h` header.
```

**Review by MathiasVP — COMMENTED** [review]
I'd like to slightly reword the change note (see my suggestion), but otherwise this LGTM! I'll start running our internal testing as well

File: cpp/ql/lib/change-notes/2023-11-8-strsafe-models.md (hunk 1 above)
Comment (bdrodes) [reply cpp/ql/lib/change-notes/2023-11-8-strsafe-models.md:4]: Whichever you prefer. I was just capturing that there were existing models there but incomplete, but I'm ok with whatever.

**Comment by bdrodes** [comment]
> I'd like to slightly reword the change note (see my suggestion), but otherwise this LGTM! I'll start running our internal testing as well

I accepted your change.

**Comment by MathiasVP** [comment]
DCA looks good (or rather: uneventful).

CI is failing because the change note is called `cpp/ql/lib/change-notes/2023-11-8-strsafe-models.md` instead of `cpp/ql/lib/change-notes/2023-11-08-strsafe-models.md`. If you rename that file this PR LGTM!

**Comment by bdrodes** [comment]
> DCA looks good (or rather: uneventful).
> 
> CI is failing because the change note is called `cpp/ql/lib/change-notes/2023-11-8-strsafe-models.md` instead of `cpp/ql/lib/change-notes/2023-11-08-strsafe-models.md`. If you rename that file this PR LGTM!

Fixed.

**Comment by MathiasVP** [comment]
Great! Now CI complains about a non-ASCII character, though 😂:
```
File "ql/cpp/ql/lib/semmle/code/cpp/models/implementations/Printf.qll" contains a non-ASCII character at the location marked with `|` in:
gCchPrintf", //StringCchPrintf(pszDest, cchDest, pszFormat, |
```

**Comment by bdrodes** [comment]
> Great! Now CI complains about a non-ASCII character, though 😂:
> 
> ```
> File "ql/cpp/ql/lib/semmle/code/cpp/models/implementations/Printf.qll" contains a non-ASCII character at the location marked with `|` in:
> gCchPrintf", //StringCchPrintf(pszDest, cchDest, pszFormat, |
> ```

Fixed, I hope.

**Review by MathiasVP — APPROVED** [review]
LGTM!

**Pull request merged**